templates/archives/stories/debugging.jinja2

```diff
@@ -1,38 +1,60 @@
+            class="figure-img img-fluid rounded"
+            alt="Part of source for the Fortran II compiler for IBM 704." />
+        <figcaption class="figure-caption">Part of source for the Fortran II compiler.</figcaption>
+    </figure>
     <br />
     One major problem that Morse identified was the input and output.  Computers back then didn’t have a screen and a
     keyboard.  Programs had to be converted to punch card by human, which is bound to introduce errors.  And
-    according to Morse, it could often take much more time for the computer to punch out the answer than it takes to
-    run the program!  <br />
+    if the output required more than printing a few numbers, the results would have to be printed to magnetic tape
+    (which blocked the computer), then converted to punched paper tape by a slow mechanical device, and then
+    converted into human-readable format by an automatic typewriter.  Or, one could display the results on a
+    cathode-ray tube, take photographs of each screen, and then manually transcribe the photographs by hand to a more
+    legible form.<br />
     <br />
-    This latter issue was especially problematic for debugging.  In modern times, debugging is simplified by printing
-    out values of variables or use of more advanced tools like gdb that allow one to view the entire state of the
-    program as it runs.  Not so in the 1950’s.  The only way to figure out what was happening inside a defective
-    program was to halt the program prematurely and slowly wait for the computer to print out all the values, and
-    hope that one of them will point to the origin of the bug.  Because the output was so slow, Philip Morse
-    suggested that one try to write one’s entire program in one go while minimizing the number of output values.
-    Modern coding practice suggests that this is a recipe for disaster, but what else can you do?  Everyone had to
-    share the computer, meaning you couldn’t hog the machine for several hours while you wrote the program chunk by
-    chunk.  <br />
+    This input/output issue was especially problematic for debugging.  In modern times, one of the first things the
+    beginner CS student learns is to inject <code>print</code> statements all throughout their code to figure
+    out where things break down.  For the more advanced, debugging is simplified by tools like gdb that allow one to
+    insert breakpoints and view the entire state of the program as it runs.  Not so in the 1950’s.  The only way to
+    figure out what was happening inside a defective program was to halt the program prematurely and slowly wait for
+    the computer to print out all the values, and hope that one of them will point to the origin of the bug.  For
+    those not familiar with coding, this is a little like trying to figure out what's wrong with a car without
+    lifting the hood up. The logical thing to do would be to write the program in sections, but computer time was
```

Comment:
This is a really great analogy for those who don't have a computer science background :)